backend/src/services/course/course.service.ts

```diff
@@ -86,14 +92,22 @@ class Course {
       case CourseHost.UDEMY:
       case CourseHost.W_UDEMY: {
         const courseData = await this.#udemyService.getByUrl(urlObject);
-        const { description, title, url } = courseData;
 
-        return this.create({
+        const { id: udemyCourseId, description, title, url } = courseData;
+
+        const course = await this.create({
           description,
           title,
           url,
           vendorKey: VendorKey.UDEMY,
         });
+
+        await this.#courseModuleService.createModulesByCourseId(
+          udemyCourseId,
+          course.id,
+        );
+
+        return course;
```

Comment:
```suggestion
        return this.#courseModuleService.createModulesByCourseId(
          udemyCourseId,
          course.id,
        );
```

Reply:
This method returns course, after adding modules, it does not return modules

Reply:
My bad